src/main/java/com/ot/restaurant/repositories/CustomerRepositoryImpl.java

```diff
@@ -0,0 +1,81 @@
+package com.ot.restaurant.repositories;
+
+import com.ot.restaurant.entities.Customer;
+import com.ot.restaurant.exceptions.CustomerIdNullPointerException;
+import com.ot.restaurant.exceptions.CustomerStatusNotActiveException;
+import constants.Status;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class CustomerRepositoryImpl implements CustomerRepository {
+  private final List<Customer> customerList = new ArrayList<>();
```

Comment:
For a litle bit of help, try to use a java `Map` in order to improve findById performance to O(1);

If you are search a litle bit further and also you want to improve sorting when finding all customers, you could use a Tree combined with a map. You should be able to look for a data structure that combines Tree benefits + Map benefits into just one data structure